tools/pylint/gevent_checker.py

```diff
@@ -24,6 +25,76 @@
     "When calling `Group.join` or `Pool.join` the flag `raise_error` must be set to "
     "`True`, otherwise exceptions will go unoticed."
 )
+INPUT_FORBIDDEN_ID = "gevent-input-forbidden"
+INPUT_FORBIDDEN_MSG = (
+    "`input` is a global function, therefore it can not be monkeypatched, "
+    "calling this function will block not only the thread using `input()` but "
+    "also the event loop, effectively bringing the process to a halt until input "
+    "is given, this is usually not the intended behavior."
+)
+SYS_STDFDS_FORBIDDEN_ID = "gevent-sys-stdfds-forbidden"
+SYS_STDFDS_FORBIDDEN_MSG = (
+    "gevent does not monkey patch the `sys.std(io,err,out)`, this means these "
+    "interfaces must not be directly used otherwise the event loop will be "
+    "halted."
+)
```

Comment:
I'm not sure this makes sense. 
No file IO is cooperative under monkeypatched gevent ([explanation why this is so](https://github.com/gevent/gevent/issues/1070)). 

Singling out `sys.std*` seems... disproportionate.

If we want to ensure cooperative file IO we would also need to prevent direct usage of `print`, `<file>.write() / .read()`, etc. and force usage of `gevent.fileobject.FileObject` for example.

Reply:
This PR is fixing the issue with waiting for user input, which in our code has been wrongly implemented with calls to `input`, and `sys.stdin`. The other files should not be a problem since the files will have data, sockets are not used through a file interface, and at least for now we don't use pipes.

Reply:
Ok, this is a rare enough edge case that I don't think it's worth to discuss. Let's just keep it.